src/IB/IBFEInstrumentPanel.cpp

```diff
@@ -0,0 +1,1067 @@
+
+#include "tbox/Pointer.h"
+#include "tbox/RestartManager.h"
+#include "tbox/SAMRAI_MPI.h"
+#include "tbox/Timer.h"
+#include "tbox/TimerManager.h"
+#include "tbox/Utilities.h"
+
+#include "libmesh/boundary_info.h"
+#include "libmesh/dense_vector.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/exodusII_io.h"
+#include "libmesh/face_tri3.h"
+#include "libmesh/linear_implicit_system.h"
+#include "libmesh/mesh.h"
+#include "libmesh/mesh_function.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/point.h"
+#include "libmesh/serial_mesh.h"
+
+#include "ibamr/IBFEMethod.h"
+#include "ibtk/FEDataManager.h"
+#include "ibtk/IndexUtilities.h"
+
+#include "BoxArray.h"
+
+#include <sstream>
+
+// static functions
+namespace
+{
+double
+linear_interp(const Vector& X,
+              const Index<NDIM>& i_cell,
+              const Vector& X_cell,
+              const CellData<NDIM, double>& v,
+              const Index<NDIM>& /*patch_lower*/,
+              const Index<NDIM>& /*patch_upper*/,
+              const double* const /*x_lower*/,
+              const double* const /*x_upper*/,
+              const double* const dx)
+{
+    boost::array<bool, NDIM> is_lower;
+    for (unsigned int d = 0; d < NDIM; ++d)
+    {
+        is_lower[d] = X[d] < X_cell[d];
+    }
+    double U = 0.0;
+#if (NDIM == 3)
+    for (int i_shift2 = (is_lower[2] ? -1 : 0); i_shift2 <= (is_lower[2] ? 0 : 1); ++i_shift2)
+    {
+#endif
+        for (int i_shift1 = (is_lower[1] ? -1 : 0); i_shift1 <= (is_lower[1] ? 0 : 1); ++i_shift1)
+        {
+            for (int i_shift0 = (is_lower[0] ? -1 : 0); i_shift0 <= (is_lower[0] ? 0 : 1); ++i_shift0)
+            {
+                const Vector X_center(X_cell[0] + static_cast<double>(i_shift0) * dx[0],
+                                      X_cell[1] + static_cast<double>(i_shift1) * dx[1]
+#if (NDIM == 3)
+                                      ,
+                                      X_cell[2] + static_cast<double>(i_shift2) * dx[2]
+#endif
+                                      );
+                const double wgt =
+                    (((X[0] < X_center[0] ? X[0] - (X_center[0] - dx[0]) : (X_center[0] + dx[0]) - X[0]) / dx[0]) *
+                     ((X[1] < X_center[1] ? X[1] - (X_center[1] - dx[1]) : (X_center[1] + dx[1]) - X[1]) / dx[1])
+#if (NDIM == 3)
+                     *
+                     ((X[2] < X_center[2] ? X[2] - (X_center[2] - dx[2]) : (X_center[2] + dx[2]) - X[2]) / dx[2])
+#endif
+                         );
+                const Index<NDIM> i(i_shift0 + i_cell(0),
+                                    i_shift1 + i_cell(1)
+#if (NDIM == 3)
+                                        ,
+                                    i_shift2 + i_cell(2)
+#endif
+                                        );
+                const CellIndex<NDIM> i_c(i);
+                U += v(i_c) * wgt;
+            }
+        }
+#if (NDIM == 3)
+    }
+#endif
+    return U;
+} // linear_interp
```

Comment:
One thing that I have _stopped_ doing is including these comments at the end of functions that give the function name. I will probably strip all of them out at some point over the summer...